### PR TITLE
Doom: show secrets on automap when revealed

### PR DESCRIPTION
**Pull request by kitchen-ace**

### CODE REVIEW

**Comment by kitchen-ace** [comment]
Sorry it took me so long to get to this. Will do Heretic soon, but wanted to make sure everything's good with the Doom version first.

**Comment by JNechaevsky** [comment]
Oh, I was afraid that it will require extra menu item. 🙂 Can it be merged now, or Heretic is also planned?

**Comment by JNechaevsky** [comment]
Absolutely no problem with time! Looking at changelog, there are just a few improvements since last release:

<details>

### Improvements
* Save and Load menu now showing date and time of savegame file (thanks fabiangreffrath).
* Added `savegames_path`, `screenshots_path` config variables and `-shotdir` (thanks liPillON).
* One-sided automap lines now drawing last (Woof! issue [#1665](https://github.com/fabiangreffrath/woof/issues/1665). Thanks ribbiks, rfomin and fabiangreffrath).


### Updates
* Support for Windows XP is back.
* Updated SDL library to version 2.30.2 and FluidSynth to version 2.3.5
* Replaced MinGW libraries with official releases of SDL and FluidSynth to provide compatibility with WinXP and reduce amount of libraries in general.

</details>

There is simply nowhere to hurry up. So please, feel free to take your time as much as needed.

**Comment by kitchen-ace** [comment]
Heretic is planned, but I probably won't be able to get to it for another day or two at least. So you can merge this now if you want. (I wanted to be sure there weren't going to be any merge conflicts, so definitely merge it now if you're planning on doing any other map-related changes in the near future.)

File: src/doom/am_map.c [hunk 1]
```diff
@@ -1766,10 +1769,15 @@ static void AM_drawWalls (void)
...
                             // [JN] Highlight secret sectors
-                            if (automap_secrets && lines[i].frontsector->special == 9)
+                            if (automap_secrets > 1 && lines[i].frontsector->special == 9)
                             {
                                 AM_drawMline(&l, secretwallcolors);
                             }
+                            // [plums] show revealed secrets
+                            else if (automap_secrets && lines[i].frontsector->oldspecial == 9)
+                            {
+                                AM_drawMline(&l, foundsecretwallcolors);
```
Comment (JNechaevsky) [inline src/doom/am_map.c:1779]: Uh-oh, there might be a problem with merging. Have a look please, I recently [updated](https://github.com/JNechaevsky/international-doom/commit/385c0700896f7a34143ceae00661827822a06336) walls drawing function to use `array_push` in all three games: https://github.com/JNechaevsky/international-doom/blob/main/src/doom/am_map.c#L1779-L1782

Comment (kitchen-ace) [reply src/doom/am_map.c:1779]: Yeah that will definitely be a problem. I had hoped github would have flagged that as a merge conflict, but it didn't. 

I'll address that soon, thanks for checking it.

**Comment by kitchen-ace** [comment]
I'm going to close this for now and then rewrite it against the current codebase in a few days. You should merge #129 in the meantime.

**Pull request closed without merging**